java/src/jmri/jmrit/throttle/FunctionButton.java

```diff
@@ -572,6 +572,24 @@ public boolean isImageOK() {
     public boolean isSelectedImageOK() {
         return isSelectedImageOK;
     }
+    
+    private int _maxFunc;
+    
+    /** 
+     * Set maximum function number hardware is capable of.
+     * @param maxfunc normally 29, 0-28.
+     */
+    public void setMaxFunction( int maxfunc) {
+        _maxFunc = maxfunc;
+    }
+    
+    /**
+     * Get maximum function number hardware is capable of.
+     * @return maximum functions, normally 29, i.e. 0-28.
+     */
+    public int getMaxFunction() {
+        return _maxFunc;
+    }
```

Comment:
I find the name MaxFunction confusing.  The first time I read this I thought it should be 28---the maximum function number most throttles support.  Perhaps the name should refer to the number of functions: numFunction or similar.

Reply:
> @jszinger I find the name MaxFunction confusing. The first time I read this I thought it should be 28---the maximum function number most throttles support.

I agree. I'd say `MaxFunctionNum` This brings it into line with work I've already done in [RosterEntry](https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrit/roster/RosterEntry.java).


>  Perhaps the name should refer to the number of functions: numFunction or similar.

I think we should avoid a count-oriented variable:
- For throttles, valid Functions are F0, F1 etc.
- In Decoder definitions, `numFns` was defined to include F0(f), F0(r), F1 etc. in the Function Map. Then manufacturers started providing F0(f), F0(r), F1(f), F1(r) etc. and the concept of count became confusing.

Let's avoid similar confusion with throttles.

Reply:
Thanks for spotting, function name and javadoc comment updated.